uint8 control
---
bool success